ovn-org/ovn-kubernetes
# unittests: added unit tests for the functions in the pkg/util/ovs.go file

Signed-off-by: Vishwanath Jayaraman <vjayaram@redhat.com>



**- What this PR does and why is it needed**

This PR adds Unit Tests for the functions defined in the `go-controller/pkg/util/ovs.go` file. Most of the functions defined in the referred file directly or indirectly invoke the methods of interfaces `Interface` and  `Cmd` defined in k8s.io/utils/exec/exec.go file. Mocks for the referred interfaces needed for unit testing are generated in `go-controller/pkg/testing/mocks` directory using `mockery` from https://github.com/vektra/mockery. The table driven unit tests are written using the https://github.com/stretchr/testify/ framework.

**- Special notes for reviewers**

The PR adds just the unit tests (where mocking of the underlying  interfaces of the third party library is possible) in the ovs_unit_test.go file with minimal refactor of the code(only where necessary). Any suggestions for refactor of the actual code will need to be done in a different PR and is not in scope of this PR. Basically, would need Unit Tests with some coverage before we can do any further refactor safely.
References:
https://tutorialedge.net/golang/improving-your-tests-with-testify-go/#a-mocking-example
https://blog.codecentric.de/2019/07/gomock-vs-testify/ 
https://gowalker.org/github.com/stretchr/testify/mock#Arguments 

**- How to verify it**

Pull this PR to your project workspace
navigate to ovn-kubernetes/go-controller/pkg/util directory
execute `go test ./ovs.go ./ovs_unit_test.go -coverprofile cp.out` (@trozet thanks for the corrected command)
execute `go tool cover -html=cp.out` (a browser will be launched with the coverage report displayed, check the coverage for the file ovs.go)

**- Description for the changelog**
